### PR TITLE
some improvements to FormItem

### PR DESCRIPTION
**Pull request by claneo**
1. automatically set name for children of FormItem
1. automatically set label and required for FormItem if using yup
1. allow pass function as children to FormItem

### CODE REVIEW

File: src/Form.tsx [hunk 1]
```diff
@@ -3,14 +3,17 @@ import * as React from "react";
...
 
-export function Form(props: FormProps) {
-  return (<Field>
-    {({ form: { handleReset, handleSubmit } }: FieldProps) => (
-      <$Form onReset={handleReset} onSubmit={handleSubmit} {...props} />
-    )}
-  </Field>)
-};
+export const Form = (props: FormProps) =>
+  name ? (
```
Comment (jannikbuschke) [inline src/Form.tsx]: `name` here should come from the props object in line 8. So `props.name` or destructuring `props`

**Comment by jannikbuschke** [comment]
@claneo I am a bit afraid to approve these changes as they make this library more complex (everycomponent has if else name structures) and also now we have a hard dependency on yup.
This might not seem to be a big deal, but in my eyes simplicity itself is something intrinsincally good.

I am too not so happy with the FormItem component as the duplicate name thing is ugly, but also the FormItem component itself has to many features and it seems there are also accessability issues.

I am thinking about a second library that comes with more opinionated controls, there your yup integration could make sense.

Im also thinking about re-exporting all components wrapped insight a FormItem component, exposing both props. This way the core components simplicity remains and the more complex components are just composed of smaller ones.

**Comment by claneo** [comment]
Except to pass Typescript compile, another reason of why I add if else to every component is I don't have to import same component from both of this package and antd and rename one of them.

I'm looking for a way to change yup to an optional dependency.

My team has been working with this fork for two months, and it work fine.

I'm not sure whether this changes are good or not, maybe it can be a short-term solution while there's no breaking changes. (BTW, I think the last version 1.3.0 should be a major version update as it changes how component be imported.)

If you think this pr is not acceptable, just close it.

**Comment by jannikbuschke** [comment]
Looking at your code I think the changes work. However I am concerned with the added complexity which makes maintaining the library more difficult (and less typescript safety and yup integration its overall not acceptable for me).

Regarding v1.3: imho this version isnt a breaking change, pls respond here if otherwise https://github.com/jannikbuschke/formik-antd/issues/75

**Comment by claneo** [comment]
So how about using the style like `setFieldTouched(name || '', true)` to pass typescript compile instead of if else statement.

yup integration can be removed because I can use a modified version of `FormItem` in my project. 

Because of typescript, changing definition of `FieldProps` influenced all components, which really annoyed me.

**Comment by jannikbuschke** [comment]
Regarding typescript I am talking about the consumers of the package (not the library internals).
A user who just uses one of the core components could think that the `name` prop is optional, but it isnt. Its only optional when wrapped in a FormItem. So your proposed changes makes the api less strict.

**Pull request closed without merging**